### PR TITLE
Refactor/#40 event component

### PR DESCRIPTION
**Pull request by Juanse11**
Mostly refactoring. Deconstructed event components using  two customs Hooks: useCheckbox for handling events related to checkbox changes on tables, and useRowAction for handling events performing changes on rows, using the edit toolbar. Both of these hooks are being used on the "My events" and "Event details", substantially reducing boilerplate that was unnecessarily being repeated on both components.

Refactoring also includes changes on mobile experience.

Fixed CORS problem. URL on client .env.development must be set to http://localhost:4000/graphql.

### CODE REVIEW

File: client/src/App.jsx [hunk 1]
```diff
@@ -8,10 +8,10 @@ import { InMemoryCache } from 'apollo-cache-inmemory';
 import { setContext } from 'apollo-link-context';
 import { onError } from 'apollo-link-error';
 import { createUploadLink } from 'apollo-upload-client';
-
+import { MuiPickersUtilsProvider } from '@material-ui/pickers';
```
Comment (acjesusr) [inline client/src/App.jsx]: I belive that this dependency has been moved to home, check for it before merging

File: client/src/App.jsx [hunk 2]
```diff
@@ -8,10 +8,10 @@ import { InMemoryCache } from 'apollo-cache-inmemory';
...
 import { withCookies, Cookies } from 'react-cookie';
 import moment from 'moment';
-
+import MomentUtils from '@date-io/moment';
```
Comment (acjesusr) [inline client/src/App.jsx]: Same case here, check for its call in home.jsx file in the current state of the develop branch

File: client/src/App.jsx (hunk 1 above)
Comment (sjdonado) [reply client/src/App.jsx]: Already fixed in https://github.com/sjdonado/quevent/commit/b21d0ee6cbaaf221f859d6c365bd37f68f0ad404

File: client/src/App.jsx (hunk 2 above)
Comment (sjdonado) [reply client/src/App.jsx]: You are right, I fixed in https://github.com/sjdonado/quevent/commit/b21d0ee6cbaaf221f859d6c365bd37f68f0ad404

**Pull request merged**